crates/cadenza-eval/src/eval.rs

```diff
@@ -21,6 +21,52 @@ use cadenza_syntax::{
     span::Span,
 };
 
+/// Helper function to intern a string from rowan's SyntaxText efficiently.
+///
+/// This function optimizes the interning process for the common case where
+/// text is stored as a single contiguous chunk (e.g., simple identifiers).
+///
+/// Previously we used `.to_string().as_str().into()` which allocates a String
+/// then immediately copies it into the interner. Now we optimize for single
+/// chunks while still handling multi-chunk text correctly.
+#[inline]
+fn intern_syntax_text(text: &rowan::SyntaxText) -> InternedString {
```

Comment:
Rather than having to call a function everywhere it would be nice if used `impl From<SyntaxText> for InternedString`